### PR TITLE
feat: Adopt plugin interface proposed by gh:interledger/rfcs/pull/177

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

File: src/lib/plugin.js [hunk 1]
```diff
@@ -410,8 +413,7 @@ class FiveBellsLedger extends EventEmitter2 {
       throw new ExternalError('Unable to determine current balance')
     }
     const ledgerBalance = new BigNumber(res.body.balance)
-    const integerBalance = ledgerBalance.shift(this.ledgerContext.getInfo().scale)
-    return integerBalance.toString()
+    return ledgerBalance.toString()
```
Comment (sharafian) [inline src/lib/plugin.js]: Why was this changed? The ledger keeps amounts as decimals, but getBalance should return an integer

**Review by sharafian — CHANGES_REQUESTED** [review]

File: src/lib/plugin.js (hunk 1 above)
Comment (michielbdejong) [reply src/lib/plugin.js]: good catch, fixed. I also removed some console.log statements that were still in there.

**Comment by sharafian** [comment]
https://github.com/interledgerjs/ilp-plugin-bells/pull/118/files#diff-83847972f72b8fc383c14abb9f0791e1R417 is still using getInfo.scale

**Review by sharafian — APPROVED** [review]

File: src/lib/plugin.js [hunk 2]
```diff
@@ -379,10 +379,11 @@ class FiveBellsLedger extends EventEmitter2 {
     if (!res || res.statusCode !== 200) { throwErr() }
 
     // note that the fivebells ledger API uses 'scale' instead of 'currency_scale'.
-    // the 'ilp_prefix' is not mandatory
-    ['currency_code', 'scale', 'connectors'].map(field => {
-      if (!res.body[field]) { throwErr() }
-    })
+    // the 'ilp_prefix' field is not mandatory
+    const fields = Object.keys(res.body)
+    if (fields.indexOf('currency_code') === -1) { throwErr() }
```
Comment (sharafian) [inline src/lib/plugin.js]: instead of doing `indexOf` on the keys, you could do `if (!res.body.currency_code)`

Comment (michielbdejong) [reply src/lib/plugin.js]: Yes, that's [what I had](https://github.com/interledgerjs/ilp-plugin-bells/pull/118/commits/0128e56e36e873b895cdd3b063e7fad6a045d9e9#diff-83847972f72b8fc383c14abb9f0791e1L383) but I was causing the integration tests to fail, like in [build 485](https://circleci.com/gh/interledgerjs/ilp-plugin-bells/485), with:

```
mark 2017-03-31T14:39:42.008Z connector:app error ExternalError: Unable to determine ledger metadata
mark     at throwErr (/home/ubuntu/ilp-plugin-bells/integration-test/node_modules/ilp-plugin-bells/src/lib/plugin.js:366:13)
mark     at map.field (/home/ubuntu/ilp-plugin-bells/integration-test/node_modules/ilp-plugin-bells/src/lib/plugin.js:384:31)
mark     at Array.map (native)
mark     at FiveBellsLedger._fetchLedgerMetadata (/home/ubuntu/ilp-plugin-bells/integration-test/node_modules/ilp-plugin-bells/src/lib/plugin.js:383:46)
mark     at next (native)
mark     at onFulfilled (/home/ubuntu/ilp-plugin-bells/integration-test/node_modules/co/index.js:65:19)
mark     at process._tickCallback (internal/process/next_tick.js:103:7)
ledger:3001 2017-03-31T14:39:42.010Z ledger:koa info --> GET / 200 6ms 804B
```

I think either there's a typo in that line, or Array.map is not supported by the node version that runs our integration tests. With this change, the integration tests get past that point, like in [build 489](https://circleci.com/gh/interledgerjs/ilp-plugin-bells/489), so I'll leave it like this for now while I fix the rest of the integration tests, and then switch it back when I find out why this is happening. :)

Comment (sharafian) [reply src/lib/plugin.js]: maybe one of the fields is equal to zero? you could check for equality to undefined instead

**Comment by michielbdejong** [comment]
rebased and squashed.

File: src/lib/plugin.js [hunk 3]
```diff
@@ -379,8 +379,8 @@ class FiveBellsLedger extends EventEmitter2 {
     if (!res || res.statusCode !== 200) { throwErr() }
 
     // note that the fivebells ledger API uses 'scale' instead of 'currency_scale'.
-    // the 'ilp_prefix' field is not mandatory
-    if (!res.body.currency_code || !res.body.scale) { throwErr() }
+    // fields like 'ilp_prefix' and 'currency_code' are not mandatory
+    if (!res.body.connectors || !res.body.scale) { throwErr() }
```
Comment (michielbdejong) [inline src/lib/plugin.js]: @dappelt this was making the 'index' integration test fail (since the currency_code of `test1.ledger1` is null there), is that not something you ran into?

Comment (dappelt) [reply src/lib/plugin.js]: @michielbdejong on the branch of the ILP kit I am working on all integration tests [pass](https://circleci.com/gh/interledgerjs/ilp-kit/1646). Maybe this version of the plugin is not yet used by ILP kit?

Comment (michielbdejong) [reply src/lib/plugin.js]: ah, I think I know why. I wasn't using the `da-2.0` branch of five-bells-integration-test. Created a separate Dockerfile now which is rebased on da-2.0: https://github.com/michielbdejong/five-bells-integration-test/tree/mj-add-dockerfile-2.0

Comment (michielbdejong) [reply src/lib/plugin.js]: ah right, and I also needed to name that branch `mj-currency_scale` and push it to origin of course, so that [circleci can use it](https://circleci.com/gh/interledgerjs/five-bells-integration-test/407).

**Comment by emschwartz** [comment]
Commit was included in https://github.com/interledgerjs/ilp-plugin-bells/pull/120

**Comment by michielbdejong** [comment]
ah, thanks. sorry for leaving that dangling on friday.
